### PR TITLE
Convert to ReScript syntax

### LINKED ISSUE #17: Convert to rescript syntax
This might be easy, the question is whether there are edge cases. Also what do to about `bsdoc` - I have no idea whether that requires reason syntax.

Step one is to switch the dependency from `bs-platform` to `rescript` (this also means changing the `bsb` scripts in package.json)

From there, run `yarn rescript convert -all` according to this page:
https://rescript-lang.org/docs/manual/latest/migrate-from-bucklescript-reason

If it then compiles, the final stage to convert is the `refmt-all` script in package.json (rescript has a format command, so this shouldn't be difficult either).

### PR DESCRIPTION
**Pull request by Minnozz**
- [x] Change dependency from bs-platform to rescript
- [x] Update package.json scripts
- [x] `yarn rescript convert -all` to convert syntax
- [x] `yarn rescript` to recompile tests
- [x] Check converted code manually
- [x] Figure out `bsdoc` situation

Fixes #17

### CODE REVIEW

**Comment by TheSpyder** [comment]
Thanks for this! It will take a bit of time to review as I'll check over the files manually.

**Comment by Minnozz** [comment]
@TheSpyder Let me know if you need me to make any changes :)

**Review by TheSpyder — APPROVED** [review]
Seems fine at a cursory glance, I think the only way we'll know for sure is by using it :)

File: src/Webapi.res [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+module Dom = Webapi__Dom
+module File = Webapi__File
+
+@ocaml.doc(" Re-export from [bs-fetch] for convenience. To use, you will also
+    need to add the [bs-fetch] package as a dependency.
+
+    To get the [FormData] of an HTML form, use
+    [Webapi.Dom.HtmlFormElement.data].
+
+    @since 0.18.0 ")
```
[the remark this answers is not in the document]
Comment (Minnozz) [reply src/Webapi.res:15]: There are 14 `ocaml.doc` attributes in the converted code.

ReScript will introduce a `/** */` doc comment syntax (which will be `@ocaml.doc` attributes in the AST) in the future, which will make the conversion smoother, but it currently doesn't exist yet.

[the remark this answers is not in the document]
Comment (Minnozz) [reply src/Webapi.res]: I converted them back to `/** */` manually.

[the remark this answers is not in the document]
Comment (TheSpyder) [reply src/Webapi.res]: oh thank you!

**Comment by TheSpyder** [comment]
I'll publish a release later this week

**Pull request merged**